pkg/cloud/baremetal/actuators/machine/actuator_test.go

```diff
@@ -1630,3 +1631,208 @@ func TestDeleteOfBareMetalHostDeletesMachine(t *testing.T) {
 		}
 	}
 }
+
+func TestRemediation(t *testing.T) {
+	machine, machineNamespacedName := getMachine("machine1")
+	host, hostNamespacedName := getBareMetalHost("host1")
+	node, nodeNamespacedName := getNode("node1")
+	linkMachineAndNode(machine, node)
+	host.Status.PoweredOn = true
+
+	//starting test with machine that needs remediation
+	machine.Annotations = make(map[string]string)
+	machine.Annotations[HostAnnotation] = host.Namespace + "/" + host.Name
+	machine.Annotations[externalRemediationAnnotation] = ""
+
+	scheme := runtime.NewScheme()
+	clusterapis.AddToScheme(scheme)
+	bmoapis.AddToScheme(scheme)
+	corev1.AddToScheme(scheme)
+
+	c := fakeclient.NewFakeClientWithScheme(scheme)
+
+	actuator, err := NewActuator(ActuatorParams{
+		Client: c,
+	})
+	if err != nil {
+		t.Error(err)
+	}
+
+	c.Create(context.TODO(), machine)
+	c.Create(context.TODO(), host)
+	c.Create(context.TODO(), node)
+
+	err = actuator.Update(context.TODO(), nil, machine)
+	if err != nil {
+		t.Errorf("unexpected error %v", err)
+	}
+
+	host = &bmh.BareMetalHost{}
+	c.Get(context.TODO(), hostNamespacedName, host)
+	if !hasPowerOffRequestAnnotation(host) {
+		t.Log("Expected reboot annotation on the host but none found")
+		t.Fail()
+	}
+
+	host.Status.PoweredOn = false
+	c.Update(context.TODO(), host)
+
+	err = actuator.Update(context.TODO(), nil, machine)
+	if err != nil {
+		t.Errorf("unexpected error %v", err)
+	}
+
+	machine = &machinev1.Machine{}
+	c.Get(context.TODO(), machineNamespacedName, machine)
+
+	if _, exists := machine.Annotations[poweredOffForRemediation]; !exists {
+		t.Log("Expected powered-off-for-remediation annotation to exist on machine but none found")
+		t.Fail()
+	}
+
+	err = actuator.Update(context.TODO(), nil, machine)
+	if err != nil {
+		switch err.(type) {
+			case *clustererror.RequeueAfterError:
+				break
+			default:
+				t.Errorf("unexpected error %v", err)
+		}
+	}
+
+	node = &corev1.Node{}
+	err = c.Get(context.TODO(), nodeNamespacedName, node)
+	if !errors.IsNotFound(err) {
+		t.Log("Expected node to be deleted")
```

Comment:
Can Get() return any other error at this point? So that err != nil but is also not of the type we expect?

Reply:
Not sure I understand the question. `Get()` may return NotFound error or other errors. We only allow NotFound, as this is what we expect

Reply:
OK, the first time I read through this I thought "what if we don't get an error at all?" but looking again it's clear that would correctly cause a test failure.